### PR TITLE
[GH-184]: Use UserUsername in Link, Fixes GitLab id storage

### PR DESCRIPTION
**Pull request by n-thumann**
#### Summary

This PR aims to solve the issue with broken account links in GitLab notifications.
It basically replaces `UserName` with `UserUserName` to use the actual GitLab username instead of the displayed name in the link. While fixing that, it was also required to fix the storage of the GitLab user id. Before this commit, `userID` and `GitlabUserID` have been mixed up, so that `GetUsernameByID` wasn´t returning the correct username.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-plugin-gitlab/issues/184

### CODE REVIEW

**Comment by n-thumann** [comment]
By the way, I changed the notification format to `[n-thumann](link) has pushed 1 commit to [n-thumann/test](link)` (notice the GitLab username in the beginning instead of name) to match the other notifications, although I would personally find it better to use the actual name (so `[Nicolas Thumann](link) has pushed`), but in all notifications to be consistent. Let me know if you agree, I´ll change it then ✌️

File: server/command.go [hunk 1]
```diff
@@ -208,14 +208,16 @@ func (p *Plugin) ExecuteCommand(_ *plugin.Context, args *model.CommandArgs) (*mo
...
 					return p.getCommandResponse(args, "Unknown error please retry or ask to an administrator to look at logs"), nil
 				}
-			} else {
-				if err := p.API.KVDelete(info.GitlabUsername + GitlabUsernameKey); err != nil {
-					p.API.LogError("can't delete GitLab username in kvstore", "err", err.Error())
+				if err := p.storeGitlabIDToUserIDMapping(info.GitlabUsername, info.GitlabUserID); err != nil {
+					p.API.LogError("can't store GitLab to GitLab id mapping", "err", err.Error())
 					return p.getCommandResponse(args, "Unknown error please retry or ask to an administrator to look at logs"), nil
 				}
+			} else if err := p.API.KVDelete(info.GitlabUsername + GitlabUsernameKey); err != nil {
```
Comment (levb) [inline server/command.go:218]: 2/5 wrap this into a `p.deleteXXX` function so it's more intention-revealing and consistent with `store...`

File: server/webhook/tag.go [hunk 2]
```diff
@@ -39,7 +39,7 @@ func (w *webhook) handleDMTag(event *gitlab.TagEvent) ([]*HandleWebhook, error)
 }
 
 func (w *webhook) handleChannelTag(event *gitlab.TagEvent) ([]*HandleWebhook, error) {
-	senderGitlabUsername := event.UserName
+	senderGitlabUsername := w.gitlabRetreiver.GetUsernameByID(event.UserID)
```
Comment (levb) [inline server/webhook/tag.go:42]: Is this needed because `event.UserUsername` does not come in the message?

Comment (n-thumann) [reply server/webhook/tag.go:42]: Yes, it is :) See [here](https://github.com/xanzy/go-gitlab/blob/master/event_webhook_types.go#L86) or [here](https://docs.gitlab.com/ce/user/project/integrations/webhooks.html#tag-events)

File: server/command.go (hunk 1 above)
Comment (n-thumann) [reply server/command.go:218]: Sure, thanks :)

Comment (n-thumann) [reply server/command.go]: Done, I also refactored two other similar functions so that it´s more consistent :v:

**Review by levb — APPROVED** [review]
LGTM, thx for the improvement and the cleanup!

**Comment by jfrerich** [comment]
@iomodo reminder to review this PR.  Thanks for the contribution, @n-thumann!

**Review by iomodo — CHANGES_REQUESTED** [review]
Thanks a lot @n-thumann, great work!
One minor comment below regarding the error wrapping.

**Comment by n-thumann** [comment]
I appreciate your feedback, @iomodo 👏 I just incorporated your comments. Let me know if they´re okay now :)

**Review by iomodo — APPROVED** [review]
LGMT! Thanks @n-thumann

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Subscription posts for push events now have working link
- Post structure and link is consisted with that of other events
- Checked other events to ensure they all seem tp have working links

LGTM!

Thanks @n-thumann for this fix!

**Pull request merged**